src/app/PubSubVitessceGrid.js

```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import VitessceGrid from 'vitessce-grid';
 
 import { LayerPublisher } from '../components/layerpublisher';
+import { getPreferredTheme } from '../components/utils';
```

Comment:
This is nifty, but I have mixed feelings about it. I feel like projects generally have a solid sense of their own style, and having isolated components swapping their contrast for some users because of a browser setting would be unexpected -- I think that if this is used, it should be at a higher level and passed down.

Do you have a sense of how likely users are to twiddle this in their browser? Is it primarily an accessibility setting?

Down below you do let it be overridden by the config, but I think that's going to be tied more closely to the data... I think either a boolean or a theme-name component prop may be better.

Reply:
Very true. I will get rid of the default-to-user-preference stuff. Instead, how do you feel about the following:
- default to dark theme
- show toggle on welcome page to change the user preference between light/dark
    - save this to localstorage?
- dataset config `theme` property overrides both the default and the toggle setting
- url parameter overrides all three (default dark, toggle setting, theme property)

Reply:
I probably wouldn't cascade through so many options, but it's subjective. I'd be happy with a react prop that is set from a url parameter, and I'm not sure we have a story than requires anything more complicated than that. The code here is just a demo, and if the downstream user want to use localstorage, or put an extra setting in the config, they are free to do so... I'm just not sure we want to put cost of maintaining those options in this library.

... So, to wrap up, I think it'd be better to keep it simpler, but if you've got a compelling use case in mind, just go for it: No need for a long thread here. Thanks!